### PR TITLE
Add new dependency language parsers

### LINKED ISSUE #2289: New Dependency Language Parsers
1. Rust
2. Go
3. Kotlin

Code location for examples. 

https://github.com/chaoss/augur/tree/main/augur/tasks/git/dependency_tasks/dependency_util

### PR DESCRIPTION
**Pull request by c0d33ngr**
**Description**
- Addition of new dependency language parser for Go, Rust and Kotlin.

This PR fixes #2289 

**Notes for Reviewers**

**Signed commits**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: augur/tasks/git/dependency_tasks/dependency_util/dependency_calculator.py [hunk 1]
```diff
@@ -20,6 +23,9 @@ def get_dependency_analysis_module_tuples():
 	yield java_deps, 'java'
 	yield cpp_deps, 'C++'
 	yield c_deps, 'C'
+    yield go_deps, 'go'
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/dependency_calculator.py:26]: It looks like there is an indentation error for these three lines that need to be corrected.

File: augur/tasks/git/dependency_tasks/dependency_util/go_deps.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import sys
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/go_deps.py:1]: Does this handle cases, if applicable, where several imports are made on the same line?

File: augur/tasks/git/dependency_tasks/dependency_util/go_deps.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import re
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/go_deps.py:2]: Does this handle cases, if applicable, where several imports are made on the same line?

File: augur/tasks/git/dependency_tasks/dependency_util/go_deps.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from pathlib import Path
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/go_deps.py:3]: Does this handle cases, if applicable, where several imports are made on the same line?

File: augur/tasks/git/dependency_tasks/dependency_util/kotlin_deps.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import sys
+import re
+from pathlib import Path
+
+def get_files(path):
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/kotlin_deps.py:5]: Does this handle cases, if applicable, where several imports are made on the same line?

File: augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import sys
+import re
```
Comment (sgoggins) [inline augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py:2]: Does this handle cases, if applicable, where several imports are made on the same line?

**Review by sgoggins — CHANGES_REQUESTED** [review]
Thank you!  This looks good. I have one observation about an indentation error, and a question about "Does this handle cases, if applicable, where several imports are made on the same line?" in each case.

File: augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py (hunk 6 above)
Comment (sgoggins) [reply augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py:2]: We don't *think* rust allows multiple imports on one line like Python. We don't know about the other languages.

**Comment by sgoggins** [comment]
@c0d33ngr : We're keen to see the questions in the review address and merge this! :)

File: augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py (hunk 6 above)
Comment (c0d33ngr) [reply augur/tasks/git/dependency_tasks/dependency_util/rust_deps.py:2]: Yes, rust don't allow multiple imports from a single line. I'm updating function

**Review by sgoggins — APPROVED** [review]

**Pull request merged**